### ISSUE TITLE
Command "craft to" stops when the chosen ingredient reaches zero and you save/load.

### ISSUE THREAD

**Issue by Daxtgnomoria** [rschurade/Ingnomia#27]
I've had some starving gnomes because i ran out of wheat during winter and when I harvested it my kitchen did not resume crafting breads, because the ingredient reached 0 and the crafting order got reset after loading this game.

It was the only ingredient available to craft breads (my only type of grain), the order to craft breads was specifically to craft wheat breads (not "any") and after the ingredient ran out I saved and reloaded. If that happens and you do not save and reload, everything works fine.

I've seen this happen with carpenter also, where I had an order for "sticks - craft to 4 - pine planks", and after pine planks reached zero, I saved and reloaded and the order changed by itself to orange-wood, but did not resume on its own (it had the button "craft" to start it again).

The image on the right shows what happens if I reload a save after the ingredients reached zero (and it does not resume after the ingredients number increase)
![image](https://user-images.githubusercontent.com/50493006/61657151-3388ab00-acba-11e9-824a-1349385da522.png)

Using the order with "any ingredient" works fine even after saving and reloading.
![image](https://user-images.githubusercontent.com/50493006/61657274-777bb000-acba-11e9-98d6-3614bceeec4f.png)

**In short, orders using "craft to" with a specific ingredient (not "any"), resets by itself and stops if you save and reload after the chosen ingredient reaches zero.**

Here is a save test ready for this scenario, just wait until crafting is done, save and reload last save.
[8.zip](https://github.com/rschurade/Ingnomia/files/3418849/8.zip)

**Comment by Daxtgnomoria** [comment]
This also happens with the command "repeat" and "craft number"

**Comment by rschurade** [comment]
The reason is nothing in the workshop but how to item management and filter works, a item-material pair is stored when such an item is created. Since there is not wheat grain in the save it isn't created after loading. To fix that I need to add extra saving of that. Will do it later.

**Comment by rschurade** [comment]
fixed in 0.6.7.4

**Comment by Daxtgnomoria** [comment]
Now, when I craft something, all possible ingredients are shown, even if I had never seen such ingredient.

Is it intended behavior?

![image](https://user-images.githubusercontent.com/50493006/61739085-a578f700-ad83-11e9-8061-0c9d1fceb678.png)

**Comment by Daxtgnomoria** [comment]
Also, all possible material constructions are shown (when building box, for example).

![image](https://user-images.githubusercontent.com/50493006/61739592-a65e5880-ad84-11e9-898e-c2456eaa21bf.png)

**Comment by rschurade** [comment]
Well that's the result of adding all item material combinations at game
start.

**Comment by Daxtgnomoria** [comment]
Hmm, I've been experimenting with stockpiles since this change (now ALL items are show on stockpiles filtering), and I found the game to be extremely laggy, whenever I'm doing something related to stockpiles with lots of filtered items.

I built a general stockpile, to stock trashes and items to sell, and basically included most types of items. Now the game lags to a full stop for dozens of seconds whenever I'm adding or removing items from the filter list (specially types of items that have several materials and options, like "furniture" and "workshop") and whenever i'm building boxes, barrels or sacks there, and even worse, when gnomes are storing items there.

I did this several times on previous versions and never had any lag. My processor is not the best one (Core I5 3330 3.2 GHz), but this is the first time ever I've had lag while playing Ingnomia. 

Is there a way to add another filter to materials, so that only materials I've used, built or that showed in my world appear on stockpiles and workshops?

**Comment by rschurade** [comment]
Hmm guess I'll revert that then. Not a fa n of it anyway, I like the stockpile filter lean and clean with only the materials present in the world.

**Comment by Daxtgnomoria** [comment]
I like the idea of reverting it also. In terms of QoL, it's better to have a clean filter and workshop options, and just work with craft "any" to avoid the problem of canceling craft orders.

**Comment by rschurade** [comment]
The solution will be to also save the current filter state with all entries that are zero at that point. To avoid item-materials combinations disappearing on save/load.

**Comment by rschurade** [comment]
updated steam

**Comment by Daxtgnomoria** [comment]
I like the idea of items that are no longer in the world to stop showing so we can have a clean view of stockpiles and workshops. Would be possible to have only items that you have on the world or that are currently set up on a craft order to remain "active"?

That way, on longer games, after many items appeared and disappeared, the stockpile filter and workshop ingredients view remain clean.

**Comment by rschurade** [comment]
That's not really possible without considerable overhead.

**Comment by rschurade** [comment]
Fixed
